python/lsst/ts/phosim/CloseLoopTask.py

```diff
@@ -1411,16 +1409,17 @@ def generateRefCatalog(self, instName, butlerRootPath, pathSkyFile):
 config.mag_column_list=['g']
 """
             )
-        with open(catRefConfigFilename, "w") as fp:
-            fp.write('config.datasetIncludePatterns = ["ref_cat", ]\n')
-            fp.write('config.refCats = ["cal_ref_cat"]\n')
 
         runProgram(
-            f"ingestReferenceCatalog.py {catDir} {skyFilename} --configfile {catConfigFilename}"
+            f"convertReferenceCatalog {catDir} {catConfigFilename} {skyFilename} &> {catLogFilename}"
```

Comment:
This looks like a more general issue in `CloseLoopTask` - eg. https://github.com/lsst-ts/ts_phosim/blob/f55c6b27b1df4b8d0fbad541f9fb3b1934a57fd7/python/lsst/ts/phosim/CloseLoopTask.py#L881 with `f"butler write-curated-calibrations {butlerRootPath} lsst.obs.lsst.Lsst{butlerInstName}"` or https://github.com/lsst-ts/ts_phosim/blob/f55c6b27b1df4b8d0fbad541f9fb3b1934a57fd7/python/lsst/ts/phosim/CloseLoopTask.py#L1105 with `f"butler ingest-raws {butlerRootPath} {wfsExpDir}"` and then https://github.com/lsst-ts/ts_phosim/blob/f55c6b27b1df4b8d0fbad541f9fb3b1934a57fd7/python/lsst/ts/phosim/CloseLoopTask.py#L1119 `f"butler define-visits {butlerRootPath} lsst.obs.lsst.LsstComCam"` - is it possible to call these directly from python as well? If so, I can open a ticket to overhaul all places where we call `runProgram` forking a `butler` subprocess command into a direct python call.

Reply:
Also https://github.com/lsst-ts/ts_phosim/blob/f55c6b27b1df4b8d0fbad541f9fb3b1934a57fd7/python/lsst/ts/phosim/CloseLoopTask.py#L1359 `f"butler create {butlerRootPath}"` and `f"butler register-instrument {butlerRootPath} lsst.obs.lsst.LsstComCam"`

Reply:
I think to use the `Python` direct call is a great idea.

Reply:
`butler create` is a call to `lsst.daf.butler.Butler.makeRepo`. Not all the scripts have good entry points though -- we are discussing a way to layer the interface so that the click layer calls a script layer that does argument conversion (eg converting a butler root to a Butler instance) and then calling a reusable layer (thereby allowing the Butler instance to be passed through).